lib/resty/wasmx/shm.lua

```diff
@@ -346,6 +346,10 @@ local function shm_kv_set(zone, key, value, cas)
         return nil, "no memory"
     end
 
+    if rc == FFI_ABORT then
+        return nil, "cannot set value if the shm is already locked"
```

Comment:
In our FFI API we need very short error strings for two reasons: users can compare them to catch the error:
```lua
local ok, err = shm:set(...)
if err == "no memory" then

end
```

Or users will also concatenate it to an error string of their own choosing:
```lua
if not ok then
    return nil, "cannot set value in shm: " .. err
end
```

My suggestion for this specific error is to just say "locked".

Reply:
I like the simplicity of just returning `locked`.

I'm a bit concerned though that it might not be enough to communicate to the developer that `shm:set()` shouldn't be called while the shm is being iterated over using `shm:iterate_keys()` (I understand that users can explicitly lock the shm without iterating over it, but since there's no point in doing so I'm ignoring this case).

I chose a more verbose `err` because I didn't expected the user to check on its value, as it's just a message communicating an API misuse.

It's not like the message I proposed is doing a substantially better job communicating the misuse though, I agree. And I also find it difficult to communicate the actual cause of the error in a relatively short message.

I'll change it to `locked`.